beacon_node/network/src/beacon_processor/worker/sync_methods.rs

```diff
@@ -85,21 +85,18 @@ impl<T: BeaconChainTypes> Worker<T> {
         };
         // Check if a block from this proposer is already known. If so, defer processing until later
         // to avoid wasting time processing duplicates.
-        let proposal_already_known = self
+        let proposal_already_known = match self
             .chain
             .observed_block_producers
             .read()
             .proposer_has_been_observed(block.message())
-            .map_err(|e| {
-                error!(
-                    self.log,
-                    "Failed to check observed proposers";
-                    "error" => ?e,
-                    "source" => "rpc",
-                    "block_root" => %block_root
-                );
-            })
-            .unwrap_or(true);
+        {
+            Ok(is_observed) => is_observed,
+            // Both of these blocks will be rejected, so reject them now rather
+            // than re-queuing them.
+            Err(ObserveError::FinalizedBlock { .. })
+            | Err(ObserveError::ValidatorIndexTooHigh { .. }) => false,
```

Comment:
Great catch, I really failed to spot the infinite re-queuing behaviour that returning `true` here would have!

Reply:
I missed it too, during review!